master/security/network-policy-openstack.md

```diff
@@ -0,0 +1,104 @@
+
+### Features
+
+This how-to guide uses the following Calico features:
+
+- Calico predefined **OpenStack VM endpoint labels**
+- **GlobalNetworkPolicy** or **NetworkPolicy**
+
+### Concepts
+
+#### Multi-region deployments
+
+Using the OpenStack API, it is difficult to apply policy to cross-region network traffic because security groups are local to a single region. In Calico, each region in your OpenStack deployment becomes a separate Calico namespace in a single etcd datastore. With regions mapped to namespaces, you can easily define Calico network policy for communications between VMs in different regions. 
+
+#### Labels: more flexibility, greater security
+
+Calico provides predefined [VM endpoint labels]({{site.baseurl}}/{{page.version}}/networking/openstack/labels) (projects, security groups, and namespaces) for OpenStack deployments. You can use these labels in selector fields in Calico network policy to identify the VMs for allow/deny policy.
+
+#### Policy ordering and enforcement
+
+Calico network policy is always enforced before OpenStack security groups, and cannot be overridden by user-level security group configuration. 
+
+### Before you begin...
+
+- [Set up Calico for OpenStack]({{site.baseurl}}/{{page.version}}/networking/openstack/dev-machine-setup)
+- If you are using a multi-region VM deployment, [follow these extra steps]({{site.baseurl}}/{{page.version}}/networking/openstack/multiple-regions)
+
+### How to
+
+- [Restrict all ingress traffic between specific security groups](#restrict-all-ingress-traffic-between-specific-security-groups)
+- [Allow specific traffic between VMs in different regions](#allow-specific-traffic-between-vms-in-different-regions)
+
+#### Restrict all ingress traffic between specific security groups
+
+In the following example, we create a **GlobalNetworkPolicy** that is applied before any OpenStack security group policy. It prevents all ingress communication between the OpenStack **superman** and **lexluthor** projects. We use the predefined Calico VM endpoint label, **openstack-project-name**, to identify projects.
+
+```
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: deny-lexluthor-to-superman
+spec:
+  order: 10
+  selector: "projectcalico.org/openstack-project-name == 'superman'"
+  types:
+  - Ingress
+  ingress:
+  - action: Deny
+    source:
+      selector: "projectcalico.org/openstack-project-name == 'lexluthor'"
+---
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: deny-superman-to-lexluthor
+spec:
+  order: 10
+  selector: "projectcalico.org/openstack-project-name == 'lexluthor'"
+  types:
+  - Ingress
+  ingress:
+  - action: Deny
+    source:
+      selector: "projectcalico.org/openstack-project-name == 'superman'"
+```
```

Comment:
This example also exists on this page: https://docs.projectcalico.org/master/networking/openstack/labels

Reply:
Unfortunately, there is no plan in place right now to deal with this issue. That is, scrubbing existing content or deleting. Casey is working with me on a similar issue for Host Endpoints. Alex's goal for Q2 is simple: rewrite existing content. When we get through this goal, the obvious next step is clean up. We just aren't there yet.